src/v/security/mtls.cc

```diff
@@ -23,143 +22,15 @@ namespace security::tls {
 
 namespace detail {
 
-static constexpr const char* const rule_pattern{
-  R"((DEFAULT)|RULE:((\\.|[^\\/])*)\/((\\.|[^\\/])*)\/([LU]?).*?|(.*?))"};
-static constexpr const char* const rule_pattern_splitter{
-  R"(\s*((DEFAULT)|RULE:((\\.|[^\\/])*)\/((\\.|[^\\/])*)\/([LU]?).*?|(.*?))\s*([,\n]\s*|$))"};
-
-std::regex make_regex(std::string_view sv) {
-    return std::regex{
-      sv.begin(),
-      sv.length(),
-      std::regex_constants::ECMAScript | std::regex_constants::optimize};
-}
-
-bool regex_search(
-  std::string_view msg, std::cmatch& match, std::regex const& regex) {
-    return std::regex_search(
-      msg.begin(),
-      msg.end(),
-      match,
-      regex,
-      std::regex_constants::match_default);
-}
-
-bool regex_match(
-  std::string_view msg, std::cmatch& match, std::regex const& regex) {
-    return std::regex_match(
-      msg.begin(),
-      msg.end(),
-      match,
-      regex,
-      std::regex_constants::match_default);
-}
-
-constexpr std::string_view trim(std::string_view s) {
-    constexpr auto ws = " \t\n\r\f\v";
-    s.remove_prefix(std::min(s.find_first_not_of(ws), s.size()));
-    s.remove_suffix(std::min(s.size() - s.find_last_not_of(ws) - 1, s.size()));
-    return s;
-}
-
-constexpr std::optional<std::string_view> make_sv(const std::csub_match& sm) {
-    return sm.matched
-             ? std::string_view{sm.first, static_cast<size_t>(sm.length())}
-             : std::optional<std::string_view>{std::nullopt};
-}
+std::regex make_regex(std::string_view sv);
```

Comment:
question: are you trying to hide these functions?
nit: If not, function declarations should be in headers